packages/playwright/src/matchers/toMatchText.ts

```diff
@@ -69,11 +69,11 @@ export async function toMatchText(
       typeof expected === 'string'
         ? matcherHint(this, receiver, matcherName, 'locator', undefined, matcherOptions, timedOut ? timeout : undefined) +
         `Expected ${stringSubstring}: not ${this.utils.printExpected(expected)}\n` +
-        `Received string: ${printReceivedStringContainExpectedSubstring(
+        `Received string: ${receivedString.indexOf(expected) !== -1 ? printReceivedStringContainExpectedSubstring(
```

Comment:
`printReceivedStringContainExpectedSubstring` behaves weirdly if you pass:

```
 receivedString = something 
receivedString.indexOf(expected) = -1,
expected.length = something long
```

to it. Probably because it was intended only for the matching scenario.

Reply:
- Might as well do `Received:` instead of `Received string:`
- We should cover the regex case below as well.